lncfg/protocol_legacy_on.go

```diff
@@ -9,15 +9,4 @@ package lncfg
 //
 //nolint:lll
 type LegacyProtocol struct {
-	// CommitmentTweak guards if we should use the old legacy commitment
```

Comment:
I think we can go ahead and delete this file along with `protocol_legacy_off.go`.